src/interfaces/python/swig_typemaps.i

```diff
@@ -1292,16 +1292,85 @@ TYPEMAP_SPARSEFEATURES_OUT(PyObject,      NPY_OBJECT)
     PyErr_SetString(PyExc_SystemError, $1.what());
     SWIG_fail;
 }
-%rename(_kernel) kernel;
+
+%feature("nothread") _rename_python_function;
+%feature("docstring", "Renames a Python function in the given module or class. \n"
+					  "Similar functionality to SWIG's %rename.") _rename_python_function;
+
+%typemap(out) void _rename_python_function "$result = PyErr_Occurred() ? NULL : SWIG_Py_Void();"
+%inline %{
+	static void _rename_python_function(PyObject *type, PyObject *old_name, PyObject *new_name) {
+		PyObject *dict = NULL,
+				 *func_obj = NULL;
+#if PY_VERSION_HEX>=0x03000000
+		if (!PyUnicode_Check(old_name) || !PyUnicode_Check(new_name))
+#else
+		if (!PyString_Check(old_name) || !PyString_Check(new_name))
+#endif
+			{
+				PyErr_SetString(PyExc_TypeError, "'old_name' and 'new_name' have to be strings");
+				return;
+			}
+		if (PyType_Check(type)) {
+			PyTypeObject *pytype = (PyTypeObject *)type;
+			dict = pytype->tp_dict;
+			func_obj = PyDict_GetItem(dict, old_name);
+			if (func_obj == NULL) {
+				PyErr_SetString(PyExc_ValueError, "'old_name' name does not exist in the given type");
+				return;
+			}
+		}
+		else if ( PyModule_Check(type)) {
+			dict = PyModule_GetDict(type);
+			func_obj = PyDict_GetItem(dict, old_name);
+			if (func_obj == NULL) {
+				PyErr_SetString(PyExc_ValueError, "'old_name' does not exist in the given module");
+				return;
+			}
+		}
+		else {
+			PyErr_SetString(PyExc_ValueError, "'type' is neither a module or a Python type");
+			return;
+		}
+		if (PyDict_Contains(dict, new_name))
+		{
+			PyErr_SetString(PyExc_ValueError, "new_name already exists in the given scope");
+			return;
+		}
+		PyDict_SetItem(dict, new_name, func_obj);
+		PyDict_DelItem(dict, old_name);
+  }
+%}
 
 %pythoncode %{
+import sys
+
+_GETTERS = ["get",
+            "get_real",
+            "get_int",
+            "get_real_matrix",
+            "get_real_vector",
+            "get_int_vector"
```

Comment:
@avramidis this is the new way of listing getters and here is where the `get_char_string_list` has to be added (once this is merged of course)